File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Prompt user for pyramid height until they give a valid number (1–8)
    do
    {
        height = get_int("Insert height: \n");
    }
    while (height < 1 || height > 8);

    // Loop through each row of the pyramid
    for (int row = 1; row <= height; row++)
    {
        // Print leading spaces so the pyramid is right-aligned
        for (int spaces = height - row; spaces > 0; spaces--)
        {
            printf(" ");
        }

        // Print hashes for the left pyramid
        for (int hash = 1; hash <= row; hash++)
        {
            printf("#");
        }

        // Print the gap between the two pyramids (always two spaces)
        printf("  ");

        // Print hashes for the right pyramid
        for (int hash = 1; hash <= row; hash++)
        {
            printf("#");
        }

        // Move to the next line after finishing the row
        printf("\n");
    }
}

File: psets-06/mario.py
from cs50 import get_int

# Prompt user for height until valid
while True:
    height = get_int("Height: ")
    if 1 <= height <= 8:
        break

# Build the pyramids
for i in range(1, height + 1):
    # Left pyramid (right-aligned)
    spaces = height - i
    hashes = i

    print(" " * spaces + "#" * hashes + "  " + "#" * hashes)
